test/tools/Modules/HttpsListener/HttpsListener.psm1

```diff
@@ -0,0 +1,126 @@
+Class HttpsListener 
+{
+    [int]$Port
+    [System.Management.Automation.Job]$Job
+    
+    HttpsListener () { }
+
+    [string] ToString() {
+        return ('Port: {0}; Status: {1}' -f $This.Port, $This.JobStateInfo.State)
+    }
+}
+
+$HttpsListeners = @{}
+
+function Get-HttpsListener 
+{
+    [CmdletBinding(ConfirmImpact = 'Low')]
+    [OutputType([HttpsListener])]
+    param 
+    (
+        [ValidateRange(1,65535)]
+        [int[]]$Port
+    )
+
+    process 
+    {
+        if ($Port)
+        {
+            return $Script:HttpsListeners[$port]
```

Comment:
@iSazonov Ok, that certainly makes things easier. If that's the direction to go the `Get-WebListenerUrl` idea will work.